nautobot/extras/models/jobs.py

```diff
@@ -86,9 +250,13 @@ class Meta:
 )
 class JobResult(BaseModel, CustomFieldModel):
     """
-    This model stores the results from running a user-defined report.
+    This model stores the results from running a Job.
     """
 
+    job_model = models.ForeignKey(
+        to="extras.Job", null=True, blank=True, on_delete=models.SET_NULL, related_name="results"
```

Comment:
Why do we want to allow this FK to be null? Is there a use-case for unbound job results, or is this for backwards compatibility? Can you please provide some context on this decision?

Reply:
If we don't allow it to be null, then I think we'd need to use either `on_delete=models.CASCADE`, i.e. deleting a JobModel record would auto-delete all of its associated JobResults, or else `on_delete=models.PROTECT`, i.e. block deletion of a JobModel unless all of its JobResults were already deleted as a separate step. Neither of those seemed desirable to me but I'm absolutely open to discussion here!

Reply:
Yeah that works for me. If we could please ahead and make sure that it tracked in a comment and documented and I'm good!